app/src/main/java/at/roteskreuz/stopcorona/model/receivers/ExposureNotificationBroadcastReceiver.kt

```diff
@@ -33,9 +33,32 @@ import timber.log.Timber
 /**
  * Broadcast receiver for callbacks from exposure notification API.
  */
-class ExposureNotificationBroadcastReceiver : BroadcastReceiver(), KoinComponent {
+class ExposureNotificationBroadcastReceiver : BroadcastReceiver() {
 
-    private val workManager: WorkManager by inject()
+    companion object : KoinComponent {
+        private val workManager: WorkManager by inject()
+
+        fun onExposureStateUpdated(context: Context, token: String) {
```

Comment:
Moving implementation to the static block brokes Interface segregation principle. This is not a good practice. Please move it back to the `onReceive` method.

Reply:
ok